### ISSUE TITLE
Investigate what happens when a Cylc workflow is reloaded

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-ui#667]
**Describe exactly what you would like to see in an upcoming release**

Quite sure the UI will just gets broken if the workflow structure changed too much as we are not using the `reloaded` flag 

**Additional context**

From our group meeting on May 5th.

**Pull requests welcome!**

**Comment by hjoliver** [comment]
From my meeting notes:

> Check on scheduler reload, we get a new "initial data burst" - check for the reload flag in the delta

As I recall this means the scheduler sends a new initial burst (i.e. full scheduler state) after a reload.  But an initial burst is actually just a delta (containing the difference between a null state and the full state, instead of the difference since the previous delta) so the UI will have to *know* it is a burst in order to interpret it correctly.

**Comment by kinow** [comment]
@hjoliver I think if at any point, if we get a Workflow node added in the deltas.added, that contains any property with `host` or any other property besides ID & status (and the ones that are arrays such as taskProxies/jobs until we remove them from deltas.added.workflow) it means that that is an initial data burst?

Or maybe there's a better way to tell that that delta is an initial data burst? Will have a look at the query later, I have forgotten if there is any flag that indicates that is the first delta.

**Comment by hjoliver** [comment]
> I think if at any point, if we get a Workflow node added in the deltas.added, that contains any property with host or any other property besides ID & status (and the ones that are arrays such as taskProxies/jobs until we remove them from deltas.added.workflow) it means that that is an initial data burst?

That might be right, I'd have to take a look. 

Even so, for clarity it might be better to have a boolean flag to indicate initial burst or not.

**Comment by kinow** [comment]
>Even so, for clarity it might be better to have a boolean flag to indicate initial burst or not.

:+1:

**Comment by dwsutherland** [comment]
> Even so, for clarity it might be better to have a boolean flag to indicate initial burst or not.

There already is one, set by the scheduler on reload and passed into data-store re-instantiation. (`workflow.reloaded` I believe, should be in the added delta (I think))

**Comment by hjoliver** [comment]
So `workflow.reloaded` appears in the initial burst even if the workflow was not actually reloaded?  If so, that's useful but the name could be better. "full-state" or something?

**Comment by dwsutherland** [comment]
> So `workflow.reloaded` appears in the initial burst even if the workflow was not actually reloaded? If so, that's useful but the name could be better. "full-state" or something?

We can call it whatever we like, but yes we could generalize to all new bursts.. However, given our conversation on Element (where any `deltas.added.workflow` is considered an initial burst), we may not need it.. It is helpful when not using the `deltas` entry point though.

**Comment by kinow** [comment]
Well, at the moment the UI breaks when you call the Reload mutation (using the toolbar top menu). I saw that one delta comes with `deltas.added.worfklow`, but no status. At the moment we are filtering for `status == RUNNING`, which worked for when you restarted a workflow, or stopped and started it.

I removed that `if`, but the deltas are still not applied :confused:

**Comment by kinow** [comment]
From Element:

Huh, was testing the UI for `cylc reload` commands.
- Synced branches
- Started UI `master` and UIS `cylc hub` also `master`
- Started `five`
- Used mutations menu to reload the workflow (here I had removed an `if status === RUNNING` to fix an issue I found earlier today while testing this) (no errors in the cylc hub console or UI)
- Received multiple (3) deltas that looks the same:
```js
{
  "id": "2",
  "type": "data",
  "payload": {
    "data": {
      "deltas": {
        "id": "kinow|five",
        "added": {
          "workflow": {
            "id": "kinow|five",
            "name": "five",
            "owner": "kinow",
            "host": "ranma",
            "port": 43076,
            "__typename": "Workflow"
          },
          "__typename": "Added"
        },
        "updated": {
          "__typename": "Updated"
        },
        "pruned": {
          "__typename": "Pruned"
        },
        "__typename": "Deltas"
      }
    }
  }
}
```

I stopped when the workflow had removed the cycle point 1 from the UI/pool. Refreshing the UI I get the following displayed in the screenshot, which shows the exact state the workflow was when I hit "Reload".

However, by now the `cylc play` console for `five` is showing cycle points 20, 21....

Did I do something wrong when reloading the workflow (I'm not too familiar with this feature)? Or does this sound like a bug in flow and/or UIS?

![image](https://user-images.githubusercontent.com/304786/127788305-8ba4a90f-f7e8-4992-85e7-0b0353213ffd.png)

![image](https://user-images.githubusercontent.com/304786/127788314-abfcd2ff-b213-4551-9910-c90f42e5d3d2.png)
